server/src/main/java/org/opensearch/cluster/CompressionHelper.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.Version;
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.compress.CompressorFactory;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.common.io.stream.OutputStreamStreamOutput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+
+import java.io.IOException;
+
+/**
+ * A helper class to utilize the compressed stream.
+ */
+public class CompressionHelper {
+    private static final Logger logger = LogManager.getLogger(CompressionHelper.class);
+
+    /**
+     * It'll always use compression before writing on a newly created output stream.
+     * @param writer Object which is going to write the content
+     * @param nodeVersion version of cluster node
+     * @param streamBooleanFlag flag used at receiver end to make intelligent decisions. For example, ClusterState
+     *                          assumes full state of diff of the states based on this flag.
+     * @return reference to serialized bytes
+     * @throws IOException if writing on the compressed stream is failed.
+     */
+    public static BytesReference serializedWrite(Writeable writer, Version nodeVersion, boolean streamBooleanFlag) throws IOException {
```

Comment:
dont create static methods, it makes testing and mocking difficult.

Reply:
Mockito can now mock static method easily and we use 5.x version of Mockito. All utility classes where no instance related information is required, we don't actually create the object of those classes and simply use static methods. Footprint is also smaller for static methods. 
I'll still see if there is any scope of keeping instance variables inside the helper class, if not I prefer to keep methods static.